src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -559,13 +560,39 @@ private static Details getDetails(DetailAST ast) {
                 case TokenTypes.LITERAL_DO:
                     details = getDetailsForDoLoops(ast);
                     break;
+                case TokenTypes.LITERAL_SWITCH:
+                    details = getDetailsForSwitch(ast);
+                    break;
                 default:
                     details = getDetailsForOthers(ast);
                     break;
             }
             return details;
         }
 
+        /**
+         * Collects details about switch statements and expressions.
+         *
+         * @param switchNode switch statement or expression to gather details about
+         * @return new Details about given switch statement or expression
+         */
+        private static Details getDetailsForSwitch(DetailAST switchNode) {
+            final DetailAST lcurly = switchNode.findFirstToken(TokenTypes.LCURLY);
+            final DetailAST rcurly = switchNode.getLastChild();
+            DetailAST nextToken = null;
+            DetailAST parent = switchNode;
+            while (nextToken == null) {
+                nextToken = parent.getNextSibling();
+                if (nextToken != null && (nextToken.getType() == TokenTypes.SEMI
+                        || nextToken.getType() == TokenTypes.RPAREN
+                        || nextToken.getType() == TokenTypes.EMPTY_STAT)) {
```

Comment:
yes 

1) SEMI

lets take example lof code 
```
            int b = switch (a) {
                case 0 -> {
                    int x = 5;
                    int y = 6;
                    if (a == 2) {
                        y = 7;
                    }
                    throw new Exception();
                }
                default -> 2;
            };  --->  SEMI after Right curly
```
here the position of right curly looks valid to me and semicolon is must require here

2) Rparen
lets take example lof code 

```
      private final int field = id(switch(value) {
          case 0 -> -1;
          case 2 -> {
              int temp = 0;
              temp += 3;
              yield temp;
          }
          default -> throw new IllegalStateException();
      }); // ok   ----> Right curly position
            ```      
 The position of right curly looks valid to me it would not be good if property is alone and we put Rparen to next line

3) Empty stat
```
            try {
                switch (stValue) {
                    case 0:
                        break;
                    case 1:
                        break;
                }; // ok  ------> right curly with empty stat
            } catch (NumberFormatException e) {
                throw new IOException(" value ");
            }
```
the semi colon here is useless but if it is then the position looks valid to me 

All the three code was taken from regression 
i hope i have make you this clear

Reply:
Number 1 seems invalid.
> lets take example lof code
>            };  --->  SEMI after Right curly
> here the position of right curly looks valid to me and semicolon is must require here

https://checkstyle.org/property_types.html#RightCurlyOption
> same
> }); // this case is out of scope of RightCurly Check (see issue #5945)

I will ask @romani to confirm.

Reply:
Number 3 can be validated, but I think your understanding of this check and "looks good" is not correct. It all depends on how the check is configured, as there are 3 different options.

>             }; // ok  ------> right curly with empty stat
> the semi colon here is useless but if it is then the position looks valid to me 

Here is documented example.

https://checkstyle.org/property_types.html#RightCurlyOption
> same
>     if (a > 0) {
>        ...
>    } int i = 5; // NOT OK, no next part of a multi-block statement, so should be alone

Reply:
done
 it will skip RPAREN and SEMI